### ISSUE TITLE
REDHAWK Enterprise Integration - REDHAWK Connection

### ISSUE THREAD

**Issue by btgoodwin** [RedhawkSDR/redhawk#1]
I'm having difficulty using the REDHAWK Connection vs. the [documentation found here](https://github.com/RedhawkSDR/redhawk/releases/download/2.0.6/REDHAWK_Enterprise_v2.0.0.pdf) in REDHAWK 2.0.6 on CentOS 7.  There are a number of things that seem to be incorrect/not accounted for in the documentation vs. the karaf server.

1. If one does not specify a `domainName` in your configuration file (since it's described as optional in 5.2 Table 1), the Karaf Web Console Configuration panel, under the REDHAWK Connector service, for that configuration will auto-populate the field with `REDHAWK_DEV` even though it isn't present in the configuration file.
2. Deleting that reference (in #1) and saving the configuration adds the key to the configuration file with an empty value, and then the web console populates the field correctly.
3. If one does not specify a `domainName` when starting the service, the `redhawk.camel-redhawk` service throws an error on startup that `Domain Name was not supplied in the configuration` (in the karaf log).
4. Once the service appears to be running, only the websocket service is listed for `/redhawk` (under Apache Karaf Web Console Http).  
5. Attempting to access the REST service from `/redhawk/<nameserver>/domains`, as inferred in section 6.1.2 and 6.2.1, does not work.  The REST service appears to be working under `/cxf/redhawk/<nameserver>/domains`, however, whether one has specified a `domainName` in their configuration file or not.
6. Chapter 6 is the first mention of `nameserver` anywhere in the document with no explanation for what should be that value. If one reads to chapter 7, it can be inferred that `nameserver` is actually both the host/IP _and_ the port of the omniNames server (i.e., not just the host name/IP).
7. Per Chapter 7 (websocket), it is stated that the configured `connectionName` can be used in lieu of the `nameserver`.  This seems to also work for the REST server (which is a nice feature, please don't remove it, however it's undocumented).

## Steps:

With the omniNames service running locally (i.e., at `127.0.0.1:2809`) 

1. Install the package:

```
# yum install redhawk-enterprise-integration-dist
```

2. Create a configuration file at `$OSSIEHOME/redhawk-enterprise-integration/karaf-4.0.8/etc/redhawk.datasource.factory-config.cfg`:

```
connectionName = my-connection
port = 2809
host = 127.0.0.1
domainManager = 
deviceManagerName = 
deviceManagerFileSystemRoot = 
```

3. Start the service

```
# $OSSIEHOME/redhawk-enterprise-integration/karaf-4.0.8/bin/start
```

4. Open a web browser to: `http://localhost:8181/redhawk/127.0.0.1:2809/domains`.  Result: `HTTP ERROR 405`, `HTTP method GET is not supported by this URL`.

5. Change the url to: `http://localhost:8181/cxf/redhawk/127.0.0.1:2809/domains`.  Result: successful retrieval of the structure expected from 6.2.1.

**Comment by datadidit** [comment]
Thanks for the feedback we appreciate it and have made changes to address each point made in the upcoming 2.0.7 release of REDHAWK documentation. 

1-2: Yes the Karaf Webconsole auto-populates REDHAWK_DEV as the default Domain Manager name. If you save from the web console your configuration will have that key. If you do this via file configuration the configuration will not have that key. We've added the REDHAWK_DEV default to the REDHAWK Connector documentation. 
3. Yes, we've emphasized this point in the upcoming section of the REDHAWK Camel documentation. 
5. We've updated the base url in the 2.1.1 version of the docs. This is made even more clear in the 2.0.7 version of the docs and those changes will be pushed forward to future 2.1.X releases. 
6. Note added to REDHAWK REST chapter identifying what nameserver parameter is.
